--- io/ufile.cpp.orig	2017-02-15 13:10:43.237649600 -0500
+++ io/ufile.cpp	2017-02-15 13:03:04.066695600 -0500
@@ -20,6 +20,13 @@
 ******************************************************************************
 */
 
+/*
+ * fileno is not declared when building with GCC in strict mode.
+ */
+#if defined(__GNUC__) && defined(__STRICT_ANSI__)
+#undef __STRICT_ANSI__
+#endif
+
 #include "locmap.h"
 #include "unicode/ustdio.h"
 
